### PR TITLE
Swing sign

### PR DESCRIPTION
**Pull request by Ozzzim**
[GAME OBJECTS] [SPRITES]
## About the PR 

This fork adds a swing sign (or more accurately a people stopper, though I find that name stupid), a foldable sign that lets you write and display messages. Messages can be then read by examining the sign.

Here's a presentation video:
[www.youtube.com/watch?v=mUk8okGf_Mo](https://www.youtube.com/watch?v=mUk8okGf_Mo)
And a screenshot of the typing interface, since video capture didn't catch it.
Swingsign ui

For now the only way to get the sign is by crafting it with wood. Ideally they should be available to both catering and chapel, with perhaps a spare one in the storage/civilian quarters.

## Why's this needed? 

I feel like there's a lack of ways to advertise your stuff on the station. Your best shot is either drawing on the floor with crayons or leaving a paper which can easily be mistaken for junk or be buried under the clutter. With the swing sign Chefs will be able to show off their menu, Chaplains could write something about their sermons, Clowns could advertise their brand new show etc.

## Changelog 

```changelog
(u)Ozzzim
(*)Added a swing sign and it's closed counterpart
(+)Added new recipe to wood crafting in building_materials.dm
```

### CODE REVIEW

**Comment by DisturbHerb** [comment]
The icons don't really match perspective, and I feel like the ui could be jazzed up a little. Conceptually, I think it's neat.

File: code/obj/item/swingsign.dm [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+		..()
+		src.setItemSpecial(/datum/item_special/swipe)
+		BLOCK_SETUP(BLOCK_LARGE)
+
+	/obj/item/swingsignfolded/attack_self(mob/user as mob)
+		if(cant_drop == 1)
+			boutput(user, "You can't unfold the [src] when its attached to your arm!")
+			return
+		var/obj/swingsign/S = null
+		S = new/obj/swingsign/(user.loc)
```
Comment (TobleroneSwordfish) [inline code/obj/item/swingsign.dm:170]:
```suggestion
		var/obj/swingsign/sign = null
		sign = new(user.loc)
```
Don't use single letter variable names except for standard ones like `T` for turf, `M` for mob etc.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Neat idea, code mostly looks good, a few issues though:
- You've somehow managed to commit 4 different files related to the secret submodule, you'll need to reset them before this can be merged.
- The sprite doesn't match our perspective guidelines, see here: https://hackmd.io/@goonstation/docs/%2F%40goonstation%2Fsprites
- We have a TGUI multiline text input modal already, no need to build your own UI, see `tgui_input_text`
- Absolute path definitions like `/obj/item/swingsignfolded/attack_self` shouldn't be indented
- Comments on vars should be doc comments in the format
```
/// Comment here
var/foo = 3
```

**Comment by Ozzzim** [comment]
> * You've somehow managed to commit 4 different files related to the secret submodule, you'll need to reset them before this can be merged.

Can you tell this was my first attempt at gitting anything?

Thanks for all the input. It's bit late right now, I'll go through it properly tomorrow. I can't believe I missed the multi line input thing in the reference. Would've spared me like a week of writing.

**Review by Ozzzim — COMMENTED** [review]
Redone the static sign sprite to match the spriting standards more, recolored the folded version to match it. Held sprites have also been recolored, but aren't in yet, I just need to carve out silhouettes first.
image

File: code/obj/item/swingsign.dm [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+			if(secured)
+				boutput(user, "<span class='notice'>You unsecure [src].</span>")
+				anchored = UNANCHORED
+			else
+				boutput(user, "<span class='notice'>You secure [src].</span>")
+				anchored = ANCHORED
+			playsound(src, 'sound/items/Screwdriver.ogg', 50, 1)
+			secured = !secured
+			return
+		else if (istype(W,/obj/item/deconstructor))
```
[the remark this answers is not in the document]
Comment (Ozzzim) [reply code/obj/item/swingsign.dm:56]: I left this if case before I realized the sign doesn't really belong in the ruckingenur kit. I've removed it now

**Comment by Sovexe** [comment]
You're going to need to make a fresh branch for this

**Comment by Ozzzim** [comment]
> You're going to need to make a fresh branch for this

Yeah probably. Just to be sure, you want me to copy the branch and then make a new pull request of it, right?

**Comment by Sovexe** [comment]
I want you to make a new branch free of changes to the secret ref

**Comment by Ozzzim** [comment]
Okay done, I got a fresh branch to which I only copypasted files I've intended to change/add. Do I close this pull request and start a new one with the new branch or can I somehow change the branches for this particular merge?

I'm sorry I'm so difficult, this is a side of the Git I've never used before and I really don't want to mess up again.

**Comment by Sovexe** [comment]
Probably easiest to just make a fresh request with the new branch, and just reference it here in a comment before closing this one

**Comment by Ozzzim** [comment]
Okay I'm closing this pull and starting a different one [here](https://github.com/goonstation/goonstation/pull/16316)

**Pull request closed without merging**